craft_parts/executor/executor.py

```diff
@@ -99,6 +99,11 @@ def prologue(self) -> None:
 
         callbacks.run_prologue(self._project_info)
 
+        # obtain the stage package exclusion set.
+        packages.Repository.stage_packages_filters = (
```

Comment:
by being in the prologue this ensures that `get_stage_packages_filters()` is only called once for a given lifecycle right?

Reply:
Yes, the lifecycle prologue is executed only once for each lifecycle run. If (hypotetically) the application runs multiple lifecycles, the callback functions will run again.